utils/build-script

```diff
@@ -1473,6 +1514,191 @@ details of the setups of other systems or automated environments.""")
+            # w.r.t. testing, but it was actually unused.
+
+            # As a special case, Darwin embedded non-simulator platforms have
+            # different behavior for the test condition.
+            if stdlib_target.platform.is_darwin and \
+               stdlib_target.platform.is_embedded and \
+               not stdlib_target.platform.is_simulator:
+                if stdlib_target.platform not in platforms_to_skip_test_host:
+                    test_host_only = True
+                    test = True
+                else:
+                    test = False
+
+            name = stdlib_target.name
+            if build:
+                # FIXME: This logic doesn't make much sense, but
+                # build-script-impl had no comments.
+                if args.build_swift_stdlib_unittest_extra or \
+                       args.validation_test or args.long_test:
+                    swift_stdlib_targets.append("swift-stdlib-"+name)
+                else:
+                    swift_stdlib_targets.append("swift-test-stdlib-"+name)
+            if build_benchmark:
+                swift_benchmark_targets.append("swift-benchmark-"+name)
+                # FIXME: This probably should respect `args.benchmark`, but a
+                # typo (?) in build-script-impl meant we always would do this.
+                swift_run_benchmark_targets.append(
+                    "check-swift-benchmark-"+name)
+            if test:
+                if test_host_only:
+                    suffix = "-non-executable"
+                else:
+                    suffix = ""
+                subset_suffix = ""
+                if args.validation_test and args.long_test:
+                    subset_suffix = "-all"
+                elif args.validation_test:
+                    subset_suffix = "-validation"
+                elif args.long_test:
+                    subset_suffix = "-only_long"
+                else:
+                    subset_suffix = ""
+                swift_test_targets.append("check-swift{}{}-{}".format(
+                    subset_suffix, suffix, name))
+                if args.test_optimized and not test_host_only:
+                    swift_test_targets.append(
+                        "check-swift{}-optimize-{}".format(
+                            subset_suffix, name))
+
+        return {
+            "SWIFT_STDLIB_TARGETS": " ".join(swift_stdlib_targets),
+            "SWIFT_SDKS": " ".join(sorted(swift_sdks)),
+            "SWIFT_BENCHMARK_TARGETS": " ".join(swift_benchmark_targets),
+            "SWIFT_RUN_BENCHMARK_TARGETS": " ".join(
+                swift_run_benchmark_targets),
+            "SWIFT_TEST_TARGETS": " ".join(swift_test_targets),
+        }
+
+    # Add the options for each possible host.
+    options = {
+        args.host_target: get_options_for_target(args.host_target)
+    }
+    for target in (target for options in args.cross_compile_hosts
+                   for target in options):
```

Comment:
I meant, if we use `DeploymentTarget` in that sense, then it is valid to use `DeploymentTarget` to talk about both the host (in build/host/target) and the target (in build/host/target). So what term do we use to refer to the latter? If it is `stdlib-deployment-target`, which is what I thought you were proposing, then we don't have a term to use to describe building the stdlib for `build` (something we would need if, for example, the package manager supported cross compiling (which it will need to) and we cross compiled the package manager).